### ISSUE TITLE
Fix crashes and warnings:

### ISSUE THREAD

**Issue by juniorz** [twstrike/coyim#101]
This I believe to be associated presence changes.

```
(coyim:8577): GLib-GObject-CRITICAL **: g_value_unset: assertion 'G_IS_VALUE (value)' failed
>>> DEBUG [xxx@dukgo.com/f7d08ff8-8968-43b5-a957-cfd0f3f30fd8] Presence from xxx@dukgo.com/kingnaldo: show:  status: Disconnected: closed gone: true
```

It seems we are adding tags multiple times

```
(coyim:8577): Gtk-WARNING **: A tag named 'outgoingUser' is already in the tag table.
(coyim:8577): Gtk-WARNING **: A tag named 'incomingUser' is already in the tag table.
(coyim:8577): Gtk-WARNING **: A tag named 'outgoingText' is already in the tag table.
(coyim:8577): Gtk-WARNING **: A tag named 'incomingText' is already in the tag table.
(coyim:8577): Gtk-WARNING **: A tag named 'statusText' is already in the tag table.
```

Fix dialogs without parent

```
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
```

And this is a crash:

```
[CoyIM] 2015/10/28 12:41:44 /Users/rjunior/go/src/github.com/twstrike/coyim/session/session.go:401: -> Stanza &xmpp.ClientMessage{XMLName:xml.Name{Space:"jabber:client", Local:"message"}, From:"xxx@riseup.net/Gajim", ID:"150", To:"xxx@dukgo.com/90e4efeb-8080-4a7a-b5c1-a9bd9f6ec0d3", Type:"chat", Subject:"", Body:"?OTR:AAIDAAAAAAMAAAAEAAAAwFEQbPedjK06z+jSKQBXdr6Ip1H+igSPXq7mAjsT+ggBzocNmmdFKSJf9zw3zMNxbaNsMFA2N6lSZ8M5rAdMON0FM8q4w9HR2AB6WcO3G3/Xg6hYKEkHD6+PmV5lEm2Hui2e684VDwoZw7dfxAJLmaw5C9aGTlvX2s4iwA1IZ6Zny/7yIb3/M3mWllH0T9+6clkbpqvcnuQlVML7ulazlajoCBmu2uBZ7kL7MfHPef95bg9ywAQ1fufQ+hxsc3yzrgAAAAAAAAACAAAAFFxMsqaDCXIafk5FkRlMkrhhwMC7/7mrHmROh0soEh/ukDMnuyzG3bEAAAAojHjeofy90zwfgDv+s5zc+bTZWl+oUIWjYyP0VpNJ7WCWIqaOGUlquQ==.", Thread:"YAFVSHSWBQLrcUfBCuJumTTAitVVoPjh", Delay:(*xmpp.Delay)(nil), Extensions:xmpp.Extensions(nil)}
fatal error:
(coyim:8781): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
unexpected signal during runtime execution

(coyim:8781): GLib-GObject-WARNING **: instance with invalid (NULL) class pointer

(coyim:8781): GLib-GObject-CRITICAL **: g_signal_emit_valist: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(coyim:8781): GLib-GObject-CRITICAL **: g_object_set_qdata: assertion 'G_IS_OBJECT (object)' failed

(coyim:8781): GLib-GObject-WARNING **: instance with invalid (NULL) class pointer

(coyim:8781): GLib-GObject-CRITICAL **: g_signal_handlers_destroy: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed

(coyim:8781): GLib-GObject-WARNING **: instance with invalid (NULL) class pointer

(coyim:8781): GLib-GObject-CRITICAL **: g_signal_handlers_destroy: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
[signal 0xb code=0x1 addr=0xc0 pc=0x5988fbb]
```

**Comment by olabini** [comment]
I've now hit a segfault. =(

**Comment by olabini** [comment]
The tags make NO sense to me - AFAICS we only add them once per text buffer.

**Comment by olabini** [comment]
And I have never seen those warnings on my machine

**Comment by juniorz** [comment]
The tags, is because we are calling `outgoingUser, _ := gtk.TextTagNew("outgoingUser")` each time we create a conversation window. We should create them only once, and attach to each new conversation window.

**Comment by olabini** [comment]
No, the documentation states a tag can only be associated with one text buffer at the same time.

**Comment by juniorz** [comment]
This is when it crashes on the conversation window.

I open the conversation window and the UI is loaded the XML file.

```
[CoyIM] 2015/10/28 13:58:35 /Users/rjunior/go/src/github.com/twstrike/coyim/gui/ui_reader.go:47: Loading UI definition "ConversationDefinition" from: gui/definitions/ConversationDefinition.xml
```

I send 3 messages (see 3x MessageEventEncryptionRequired event). It probably sent 3 Query Messages.

```
[CoyIM] 2015/10/28 13:58:38 /Users/rjunior/go/src/github.com/twstrike/coyim/event/otr_event_handler.go:102: HandleMessageEvent(MessageEventEncryptionRequired, , <nil>)
[CoyIM] 2015/10/28 13:58:39 /Users/rjunior/go/src/github.com/twstrike/coyim/event/otr_event_handler.go:102: HandleMessageEvent(MessageEventEncryptionRequired, , <nil>)
[CoyIM] 2015/10/28 13:58:39 /Users/rjunior/go/src/github.com/twstrike/coyim/event/otr_event_handler.go:102: HandleMessageEvent(MessageEventEncryptionRequired, , <nil>)
```

I receive 3 DH commit. They have different `ID`


```
[CoyIM] 2015/10/28 13:58:45 /Users/rjunior/go/src/github.com/twstrike/coyim/session/session.go:401: -> Stanza &xmpp.ClientMessage{XMLName:xml.Name{Space:"jabber:client", Local:"message"}, From:"xxx@dukgo.com/kingnaldo", ID:"purpleccff1d3d", To:"strike-test@dukgo.com/bd78e281-f0c7-47d2-ad2c-194751cc92f9", Type:"chat", Subject:"", Body:"?OTR:AAICAAAAxNo0lc8sGoAjMo4KUQ4oR3GWr535ahWt4SD5FwMR8jL4vWds5Bobz8EqQH8yvLNumYsYaO5hY8PXqYvXB7pmCHZ/Epfx+fDl6sNvm4f7PNolg46xDSmQoCXKhBdYg8fA6VxVPNIAqj3s+LFY28o6c5dPjr9DOB4T7grxequjaSJAfL4FG66f2gWTI8O/oCLiFZkkR5AOBcVXELeUY7Hd0AblWN0FDfAdzGL/pCTj6yJO5/r8xq2ART2eIVlI/BhxANDfUSQAAAAg3E+BUgbUHbtZLzVX6PJUT2dUxKhXALE0CK+ehQrSN14=.", Thread:"", Delay:(*xmpp.Delay)(nil), Extensions:xmpp.Extensions(nil)}
[CoyIM] 2015/10/28 13:58:45 /Users/rjunior/go/src/github.com/twstrike/coyim/session/session.go:401: -> Stanza &xmpp.ClientMessage{XMLName:xml.Name{Space:"jabber:client", Local:"message"}, From:"xxx@dukgo.com/kingnaldo", ID:"purpleccff1d3e", To:"strike-test@dukgo.com/bd78e281-f0c7-47d2-ad2c-194751cc92f9", Type:"chat", Subject:"", Body:"?OTR:AAICAAAAxBPgC2DU4156nGVeTexvPyM2oTDOb+Fp0iDN7QsPmoBCHUgnZDJ4KfWuArG2IigLMIVvwmqdVBxOfe6jKYgahmLuc9o7gL6N0jxVxFY8fBh1m+fkyi+uGJ1yAlbBM9Ju2sxGSMYhIVw3Qshst13hQC0Wp7WK2Y7iXfkB/MQlzfxCnPG9VKOpt3/o0h8au+ek3DoCXMwu9KhA8OalLlERMKyiyLADt/sLV3QJjYuQGiIK3sHIcXUTPwRYXry5JlNrpD4h+DgAAAAgmLVq5yxpykrtBpMkJu6xbNnJfNHXaypEvEh2m+e0QLA=.", Thread:"", Delay:(*xmpp.Delay)(nil), Extensions:xmpp.Extensions(nil)}
[CoyIM] 2015/10/28 13:58:45 /Users/rjunior/go/src/github.com/twstrike/coyim/session/session.go:401: -> Stanza &xmpp.ClientMessage{XMLName:xml.Name{Space:"jabber:client", Local:"message"}, From:"xxx@dukgo.com/kingnaldo", ID:"purpleccff1d3f", To:"strike-test@dukgo.com/bd78e281-f0c7-47d2-ad2c-194751cc92f9", Type:"chat", Subject:"", Body:"?OTR:AAICAAAAxIAw/7f0D5JhFdKh/M24eCBXNqmWCEzmACGs1NGAEeq5OKtf1ycovjwUpnDdJGSSZFDFrnXrUIeGttL1ppqQbxU61/xeDo2YtPHrll/fR45+7wxw+CYdfdhsJB4oOqX/XRJ5k3PCED1L6MgwpXdu0eelGhdgIu0p6VdVFCYgYPgGAOXyS2O9Aos+k24h2kP47UPLI9vVr327iqFymCMBlgLbVj7UcoUKzppxs2PPzbGo+brQ5IZer6QmsvII5fRXk9WhyCEAAAAgfGOCX/2qailKDfgWfyaEw1g4P8cmwncBaXpJdpzz9IM=.", Thread:"", Delay:(*xmpp.Delay)(nil), Extensions:xmpp.Extensions(nil)}
```

I probably send 3 signature messages. I receive one reveal sig and the AKE finishes (see the `GoneSecure` event)

```
[CoyIM] 2015/10/28 13:58:49 /Users/rjunior/go/src/github.com/twstrike/coyim/session/session.go:401: -> Stanza &xmpp.ClientMessage{XMLName:xml.Name{Space:"jabber:client", Local:"message"}, From:"xxx@dukgo.com/kingnaldo", ID:"purpleccff1d40", To:"strike-test@dukgo.com/bd78e281-f0c7-47d2-ad2c-194751cc92f9", Type:"chat", Subject:"", Body:"?OTR:AAIRAAAAEKaTDwGmvqe3Ig6e/Op4HssAAAHSQtGz+ExUQd4n4j5MIzgwlJ5BPlfWZxjIwDfnIl0O83EW1invCz1Hx5eRCu1M4YR0F2YXckoV+EwjL+RNCKXlW38+DYxAGppJ3RyVTjC7e3qQ+4hguUXHbndpfjGWtTFK5J/LKwZQRWg6V+WFHfq0+Hqnds7gspab8Gc1Vk2MITLgVha2ONaNRj9quqyPAfWnBcJ3lGiDRtM3DgASIHeMb7sGSTWisdee6u4HRdIdWXBsomPzypY9HGEN81ouhOOe8b1zCyvx9YameKtO9RVfNm/1W+0v9X0hRmIe1zt7s8z9EhC+HffS+qRZIaBZ8doT2kt7O6Bzih4irtBQTuZqWxiWXiUPSJ53rAPui0nNOfLbUz+Y6TEtSeNANikEzyIGXZU1JKyrzRE5sHSn5cRD4p1oJhy27NQi8eFlIhU0HGOnOhpUcGffG3C0eUXH977j05uaMzc4S8Dt/pLc2YlMs9CFdHXyXdtMjL4Ze7a7DdtJO6gfc1WyE3sm7Wwohk/rmSFvl/Al++pKlRYJSwJpd/UhXcFTGVfpGmyy8alaL3T7qUmXEOKCIhIgngXyt+sAJaMzT2/3nfFlF+LdKS28IaS+3+JUmHWhEALRrg16rePuDhxMPF+K6NRTfkz/qeWWUCs0NZvG.", Thread:"", Delay:(*xmpp.Delay)(nil), Extensions:xmpp.Extensions(nil)}
[CoyIM] 2015/10/28 13:58:49 /Users/rjunior/go/src/github.com/twstrike/coyim/event/otr_event_handler.go:76: HandleSecurityEvent(GoneSecure)
```

Then I receive the other 2 reveal sig. The conversation was already secure, so no `GoneSecure` this time. In between them, I receive a notification that I have new keys for the other peer.

```
[CoyIM] 2015/10/28 13:58:49 /Users/rjunior/go/src/github.com/twstrike/coyim/session/session.go:401: -> Stanza &xmpp.ClientMessage{XMLName:xml.Name{Space:"jabber:client", Local:"message"}, From:"xxx@dukgo.com/kingnaldo", ID:"purpleccff1d41", To:"strike-test@dukgo.com/bd78e281-f0c7-47d2-ad2c-194751cc92f9", Type:"chat", Subject:"", Body:"?OTR:AAIRAAAAEKaTDwGmvqe3Ig6e/Op4HssAAAHSQtGz+ExUQd4n4j5MIzgwlJ5BPlfWZxjIwDfnIl0O83EW1invCz1Hx5eRCu1M4YR0F2YXckoV+EwjL+RNCKXlW38+DYxAGppJ3RyVTjC7e3qQ+4hguUXHbndpfjGWtTFK5J/LKwZQRWg6V+WFHfq0+Hqnds7gspab8Gc1Vk2MITLgVha2ONaNRj9quqyPAfWnBcJ3lGiDRtM3DgASIHeMb7sGSTWisdee6u4HRdIdWXBsomPzypY9HGEN81ouhOOe8b1zCyvx9YameKtO9RVfNm/1W+0v9X0hRmIe1zt7s8z9EhC+HffS+qRZIaBZ8doT2kt7O6Bzih4irtBQTuZqWxiWXiUPSJ53rAPui0nNOfLbUz+Y6TEtSeNANikEzyIGXZU1JKyrzRE5sHSn5cRD4p1oJhy27NQi8eFlIhU0HGOnOhpUcGffG3C0eUXH977j05uaMzc4S8Dt/pLc2YlMs9CFdHXyXdtMjL4Ze7a7DdtJO6gfc1WyE3sm7Wwohk/rmSFvl/Al++pKlRYJSwJpd/UhXcFTGVfpGmyy8alaL3T7qUmXEOKCIhIgngXyt+sAJaMzT2/3nfFlF+LdKS28IaS+3+JUmHWhEALRrg16rePuDhxMPF+K6NRTfkz/qeWWUCs0NZvG.", Thread:"", Delay:(*xmpp.Delay)(nil), Extensions:xmpp.Extensions(nil)}
>>> INFO New OTR session with xxx@dukgo.com established
[CoyIM] 2015/10/28 13:58:49 /Users/rjunior/go/src/github.com/twstrike/coyim/gui/account_events.go:150: TODO: notify new keys from xxx@dukgo.com
[CoyIM] 2015/10/28 13:58:49 /Users/rjunior/go/src/github.com/twstrike/coyim/session/session.go:401: -> Stanza &xmpp.ClientMessage{XMLName:xml.Name{Space:"jabber:client", Local:"message"}, From:"xxx@dukgo.com/kingnaldo", ID:"purpleccff1d42", To:"strike-test@dukgo.com/bd78e281-f0c7-47d2-ad2c-194751cc92f9", Type:"chat", Subject:"", Body:"?OTR:AAIRAAAAEKaTDwGmvqe3Ig6e/Op4HssAAAHSQtGz+ExUQd4n4j5MIzgwlJ5BPlfWZxjIwDfnIl0O83EW1invCz1Hx5eRCu1M4YR0F2YXckoV+EwjL+RNCKXlW38+DYxAGppJ3RyVTjC7e3qQ+4hguUXHbndpfjGWtTFK5J/LKwZQRWg6V+WFHfq0+Hqnds7gspab8Gc1Vk2MITLgVha2ONaNRj9quqyPAfWnBcJ3lGiDRtM3DgASIHeMb7sGSTWisdee6u4HRdIdWXBsomPzypY9HGEN81ouhOOe8b1zCyvx9YameKtO9RVfNm/1W+0v9X0hRmIe1zt7s8z9EhC+HffS+qRZIaBZ8doT2kt7O6Bzih4irtBQTuZqWxiWXiUPSJ53rAPui0nNOfLbUz+Y6TEtSeNANikEzyIGXZU1JKyrzRE5sHSn5cRD4p1oJhy27NQi8eFlIhU0HGOnOhpUcGffG3C0eUXH977j05uaMzc4S8Dt/pLc2YlMs9CFdHXyXdtMjL4Ze7a7DdtJO6gfc1WyE3sm7Wwohk/rmSFvl/Al++pKlRYJSwJpd/UhXcFTGVfpGmyy8alaL3T7qUmXEOKCIhIgngXyt+sAJaMzT2/3nfFlF+LdKS28IaS+3+JUmHWhEALRrg16rePuDhxMPF+K6NRTfkz/qeWWUCs0NZvG.", Thread:"", Delay:(*xmpp.Delay)(nil), Extensions:xmpp.Extensions(nil)}
```

Then it comes the fun part. The garbage collector unref everything from the window I have opened, and then I get the crash. I am not sure if this is a cause or a consequence of the crash.

```
[glib] Called Unref - GtkAdjustment - 0xc8200301c0
[glib] Called Unref - GtkTextBuffer - 0xc8200301b0
[glib] Called Unref - GtkEntry - 0xc820030198
[glib] Called Unref - GtkAccelGroup - 0xc820030168
[glib] Called Unref - GtkAccelGroup - 0xc820030158
[glib] Called Unref - GtkAccelGroup - 0xc820030148
[glib] Called Unref - GtkAccelGroup - 0xc820030138
[glib] Called Unref - GtkAccelGroup - 0xc820030128
[glib] Called Unref - GtkAccelGroup - 0xc820030118
[glib] Called Unref - GtkAccelGroup - 0xc820030108
[glib] Called Unref - GtkTextTag - 0xc8200300e8
[glib] Called Unref - GtkTextTag - 0xc8200300d8
[glib] Called Unref - GtkTextTag - 0xc8200300c8
[glib] Called Unref - GtkTextTag - 0xc8200300b8
[glib] Called Unref - GtkTextTag - 0xc8200300a8
[glib] Called Unref - GtkTextBuffer - 0xc820030088
[glib] Called Unref - GtkEntry - 0xc820030080
[glib] Called Unref - GtkTreeView - 0xc820030000
[glib] Called Unref - GtkAdjustment - 0xc8200861d0
[glib] Called Unref - GtkTextBuffer - 0xc8200861a8
[glib] Called Unref - GtkEntry - 0xc820086170
fatal error: unexpected signal during runtime execution
[signal 0xb code=0x1 addr=0x8 pc=0x59644a8]
```

**Comment by juniorz** [comment]
The crash seems to be fixed. There is still the `(Coy-bin:6879): GLib-GObject-CRITICAL **: g_value_unset: assertion 'G_IS_VALUE (value)' failed` we need to debug.

**Comment by juniorz** [comment]
This seems to be the reason: https://github.com/gotk3/gotk3/pull/31

**Comment by juniorz** [comment]
The only missing thing is the `Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.`

**Comment by olabini** [comment]
That one is easy - we just actually set a transient parent for modal dialogs - I've already done it for some. search for the word "transient" in gui/master_password.go

**Comment by olabini** [comment]
I have seen another GValue thing show up at times. =/

**Comment by juniorz** [comment]
All the dialogs without transient were fixed. This is closed by now, until we identify where the GValue thing happens.
